R/unname_all_chunks.R

```diff
@@ -29,8 +40,16 @@ unname_all_chunks <- function(path){
     return(invisible("TRUE"))
   }
 
-  # preserve the setup label, delete the others
-  chunk_headers_info$name[chunk_headers_info$name != "setup"] <- ""
+  if ( is.null(ch_n_p)){
+    # preserve the setup label, delete the others
+    chunk_headers_info$name[chunk_headers_info$name != "setup"] <- ""
+  } else {
+    # preserve labels not starting with ch_n_p
```

Comment:
what if the prefix is "set"? I think the setup chunk should still be preserved.

Reply:
All chunk labels will be preserved except the ones starting with the prefix. So 'setup' will be kept unless one specifies `chunk_name_prefix = 'setup'` or a shorter form as 'setu'. Or do I misunderstand?

Reply:
Could you make sure setup is never deleted?

Reply:
I clearly misunderstood ;) 
I am not sure that the name 'setup' is important: the functionality of a chunk does not change by naming it 'setup'?
But it is no problem to enforce that it is never deleted: I will do so.

Reply:
That label has at least some importance in RStudio https://github.com/rstudio/rstudio/blob/cf5076a88a219a275a2d128191c12c1f8f4e3890/src/cpp/session/modules/rmarkdown/NotebookQueue.cpp#L261-L266, not sure about elsewhere. Thanks for the PR update, I'll have a look tomorrow!

Reply:
Sooo when reading "the functionality of a chunk does not change by naming it 'setup'" I thought "of course it _does_ change!" but had trouble finding a reference so I asked for help. See https://community.rstudio.com/t/what-defines-the-setup-chunk/27595/7 tl;dr the chunk named setup has a special role in the RStudio IDE when in Notebook mode, and in Shiny prerendered documents. Now I'll review the PR. :grin: